jems-testing/src/main/java/org/dmfs/jems/hamcrest/matchers/FragileMatcher.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.hamcrest.matchers;
+
+import org.dmfs.jems.fragile.Fragile;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Description;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+
+
+/**
+ * @author marten
+ */
+public final class FragileMatcher<T> extends TypeSafeDiagnosingMatcher<Fragile<T, ?>>
+{
+    private final Matcher<? super T> mDelegate;
+
+
+    public FragileMatcher(Matcher<? super T> valueMatcher)
+    {
+        mDelegate = valueMatcher;
+    }
+
+
+    public static <T> Matcher<Fragile<T, ?>> isIntact(Matcher<T> valueMatcher)
+    {
+        return new FragileMatcher<>(valueMatcher);
+    }
+
+
+    public static <T> Matcher<Fragile<T, ?>> isIntact(T expectedValue)
+    {
+        return new FragileMatcher<>(CoreMatchers.equalTo(expectedValue));
+    }
+
+
+    public static <E extends Throwable> Matcher<Fragile<?, E>> isBroken(Class<E> exception)
+    {
+        return new FeatureMatcher<Fragile<?, E>, E>(instanceOf(exception), "broken Fragile throwing",
```

Comment:
This class serves both as `Matcher` implementation and has factory for another `Matcher` created here as anonymous class.
How about making the class util class with the factory methods and having 2 inner class `Matcher`s? The `isIntact()` can also be simpler with `FeatureMatcher` probably.

Reply:
With a `FeatureMatcher` I couldn't return `false` if the tested `Fragile` throws. I'm actually considering to replace `isBroken` with a regular implementation as well.